src/PrestaShopBundle/Resources/config/services/bundle/event_listener.yml

```diff
@@ -101,3 +101,10 @@ services:
           - '@logger'
         tags:
           - { name: kernel.event_subscriber }
+
+    prestashop.bundle.controller_stack:
+        class: PrestaShopBundle\EventListener\ControllerListener
+        arguments:
+          - '@prestashop.core.routing.controller_stack'
+        tags:
+          - { name: kernel.event_listener, event: kernel.controller, method: onKernelController }
```

Comment:
Maybe you will need a priority parameter or something similar (if possible on this event) To make sure this listener is called early in the controller workflow